src/python/pants/backend/scala/subsystems/scala_infer.py

```diff
@@ -19,3 +19,8 @@ class ScalaInferSubsystem(Subsystem):
         default=True,
         help="Infer a target's dependencies by parsing consumed types from sources.",
     )
+    force_add_siblings_as_dependencies = BoolOption(
+        "--force-add-siblings-as-dependencies",
+        default=True,
+        help="If true, add all targets generated by `scala_sources` as dependencies.",
```

Comment:
Maybe something like this?

> If true, add a dependency on all `scala_source` targets generated by the same `scala_sources` target generator.
>
> If false, the dependencies will only be added if `[scala-infer].imports` is also set to false.
>
> Setting this option to true reduces the precision of dependency information. That means that you may end up compiling more than is necessary for a particular task, and that compilation will be invalidated more frequently than actually necessary. However, setting to true may be helpful if compilation fails due to missing dependencies.

(I recommend using `softwrap()` to format this. Grep for examples)

Reply:
done